apphub/one_shot_learning/siamese_network.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import tempfile
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.keras import Model, layers, Sequential
+from tensorflow.python.keras.initializers import RandomNormal
+from tensorflow.python.keras.regularizers import l2
+from tensorflow.keras.optimizers import Adam
+
+import fastestimator as fe
+from fastestimator.op import TensorOp
+from fastestimator.trace import Trace, LRController, Accuracy, ModelSaver, EarlyStopping
+from fastestimator.schedule import LRSchedule
+from fastestimator.op.tensorop import ModelOp, BinaryCrossentropy, Minmax, Augmentation2D
+
+from fastestimator.dataset.omniglot import load_data, load_eval_data, get_batch, one_shot_trial
+
+
+class ConditionalAugmentation(TensorOp):
```

Comment:
we should make a more general Op wrapper that can make any op conditional

Reply:
possibly part of the base class?

Reply:
I implemented a general wrapper for that